x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/index/store/SearchableSnapshotDirectoryTests.java

```diff
@@ -0,0 +1,361 @@
+                }
+
+                final String repositoryName = randomAlphaOfLength(10);
+                final RepositoryMetaData repositoryMetaData =
+                    new RepositoryMetaData(repositoryName, FsRepository.TYPE, repositorySettings.build());
+
+                final BlobStoreRepository repository = new FsRepository(
+                    repositoryMetaData,
+                    new Environment(Settings.builder()
+                        .put(Environment.PATH_HOME_SETTING.getKey(), createTempDir().toAbsolutePath())
+                        .put(Environment.PATH_REPO_SETTING.getKey(), repositoryPath.toAbsolutePath())
+                        .putList(Environment.PATH_DATA_SETTING.getKey(), tmpPaths()).build(), null),
+                    NamedXContentRegistry.EMPTY, BlobStoreTestUtil.mockClusterService(repositoryMetaData));
+                repository.start();
+                releasables.add(repository::stop);
+
+                final SnapshotId snapshotId = new SnapshotId("_snapshot", UUIDs.randomBase64UUID(random()));
+                final IndexId indexId = new IndexId(indexSettings.getIndex().getName(), UUIDs.randomBase64UUID(random()));
+
+                final PlainActionFuture<String> future = PlainActionFuture.newFuture();
+                threadPool.generic().submit(() -> {
+                    IndexShardSnapshotStatus snapshotStatus = IndexShardSnapshotStatus.newInitializing(null);
+                    repository.snapshotShard(store, null, snapshotId, indexId, indexCommit, snapshotStatus, true, future);
+                    future.actionGet();
+                });
+                future.actionGet();
+
+                final RepositoriesService repositories = mock(RepositoriesService.class);
+                when(repositories.repository(eq(repositoryName))).thenReturn(repository);
+
+                final IndexSettings ephemeralIndexSettings = IndexSettingsModule.newIndexSettings("_searchable_snapshot_index",
+                    Settings.builder()
+                        .put(indexSettings.getSettings())
+                        .put(SearchableSnapshots.EPHEMERAL_INDEX_REPOSITORY_SETTING.getKey(), repositoryName)
+                        .put(SearchableSnapshots.EPHEMERAL_INDEX_SNAPSHOT_SETTING.getKey(), snapshotId.getUUID())
+                        .build());
+
+                Path tmpDir = createTempDir().resolve(indexId.getId()).resolve(Integer.toString(shardId.id()));
+                ShardId ephemeralShardId = new ShardId(new Index(indexId.getName(), indexId.getId()), shardId.id());
+                ShardPath ephemeralShardPath = new ShardPath(false, tmpDir, tmpDir, ephemeralShardId);
+
+                final DirectoryFactory factory = SearchableSnapshots.newDirectoryFactory(() -> repositories);
+                try (Directory searchableDirectory = factory.newDirectory(ephemeralIndexSettings, ephemeralShardPath)) {
```

Comment:
`searchable` seems like the wrong modifier here. Maybe `snapshotDirectory`? Also elsewhere in this test suite, and similarly for `searchableIndexInput`.

Reply:
Sure